exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -43,10 +48,9 @@ func (m *AzureManagedControlPlane) SetupWebhookWithManager(mgr ctrl.Manager) err
 
 // +kubebuilder:webhook:path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedcontrolplane,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=azuremanagedcontrolplanes,verbs=create;update,versions=v1beta1,name=default.azuremanagedcontrolplanes.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Defaulter = &AzureManagedControlPlane{}
-
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (m *AzureManagedControlPlane) Default() {
+func (m *AzureManagedControlPlane) Default(_ client.Client) {
```

Comment:
Why did we add an unused parameter here? Is it to fulfill an interface where it's needed somewhere else?

Reply:
correct https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/util/webhook/mutator.go#L32

we're just not using the client for default here, just for validate